Can't import old etherpads

**Describe the bug**
I have ~35 `.etherpad` files from ~10 years ago. I don't know the exact version used back then (though perhaps @vladikoff may) 

I just setup etherpad locally and am attempting to import them. 


But I'm running into [this `assert(authorId != null);` line](https://github.com/ether/etherpad-lite/blob/68b6aacc1f3c76988c02b3ed7bc255e6d0d9bf80/src/node/db/Pad.ts#L716) which is failing the import with ` AssertionError [ERR_ASSERTION]: The expression evaluated to a falsy value`

Honestly have no idea if it's just the one problem or.. it's a harbinger and there's a pile of backcompat issues behind this one. 

I'm open to writing my own migration script to update these old .etherpad files to something modern etherpad-lite can handle.. but... before I embark on that I wanted to see if any maintainers here have any bright ideas or useful resources. :)   

I'm using import from the Web UI, not API. 

**To Reproduce**
Here's an example .etherpad file that repros: https://gist.github.com/paulirish/da70e566b3911c90f9a0d4aaf867b8e7

Maybe related: 

* [Include Author when Importing a Document · Issue #4504 · ether/etherpad-lite](https://github.com/ether/etherpad-lite/issues/4504) 
* [Plumb author ID, add new `padDefaultContent` hook by rhansen · PR #5423 [merged] · ether/etherpad-lite](https://github.com/ether/etherpad-lite/pull/5423) that introduced the author -> authorId change

<details>
<summary> stack trace</summary>

	    at async <anonymous> (/opt/etherpad-lite/src/node/hooks/express/importexport.ts:84:7)
	    at async Object.exports.doImport (/opt/etherpad-lite/src/node/handler/ImportHandler.ts:255:28)
	    at async doImport (/opt/etherpad-lite/src/node/handler/ImportHandler.ts:156:5)
	    at async Object.exports.setPadRaw (/opt/etherpad-lite/src/node/utils/ImportEtherpad.ts:118:5)
	    at Pad.check (/opt/etherpad-lite/src/node/db/Pad.ts:716:9)
	
	[2024-12-03T12:20:57.911] [ERROR] ImportHandler - Internal error during import: AssertionError [ERR_ASSERTION]: The expression evaluated to a falsy value:
	[2024-12-03T12:20:57.910] [WARN] ImportEtherpad - (pad importtest) unsupported attributes (try installing a plugin): underline

</details>

**Server:**
 - Etherpad version:  2.2.6, git sha 0c68ddc
 - OS: docker on synology dsm
 - Is the server free of plugins:  YES.  no plugins at all.  (Though apparently i'll need plugins to add support for `list`, `strikethrough`, `underline`, and `start` )

--- a/src/node/db/Pad.ts
+++ b/src/node/db/Pad.ts
@@ -172,7 +172,7 @@ class Pad {
 
   async getRevisionAuthor(revNum: number) {
     // @ts-ignore
-    return await this.db.getSub(`pad:${this.id}:revs:${revNum}`, ['meta', 'author']);
+    return await this.db.getSub(`pad:${this.id}:revs:${revNum}`, ['meta', 'author']) ?? '';
   }
 
   async getRevisionDate(revNum: number) {

--- a/src/tests/backend/specs/ImportEtherpad.ts
+++ b/src/tests/backend/specs/ImportEtherpad.ts
@@ -171,6 +171,49 @@ describe(__filename, function () {
     }
   });
 
+  // Regression test for https://github.com/ether/etherpad-lite/issues/6785
+  describe('old .etherpad imports without author metadata', function () {
+    // Old exports (circa 2014) have revisions where meta.author is missing entirely
+    const makeOldExportNoAuthor = () => ({
+      'pad:testing': {
+        atext: {
+          text: 'foo\n',
+          attribs: '|1+4',
+        },
+        pool: {
+          numToAttrib: {},
+          nextNum: 0,
+        },
+        head: 0,
+        savedRevisions: [],
+      },
+      // Revision 0 has no meta.author — this is how very old etherpads exported
+      'pad:testing:revs:0': {
+        changeset: 'Z:1>3+3$foo',
+        meta: {
+          timestamp: 1273769875976,
+          atext: {
+            text: 'foo\n',
+            attribs: '|1+4',
+          },
+        },
+      },
+    });
+
+    it('imports without error when revision lacks meta.author', async function () {
+      await importEtherpad.setPadRaw(padId, JSON.stringify(makeOldExportNoAuthor()));
+      const pad = await padManager.getPad(padId);
+      assert.equal(pad.text(), 'foo\n');
+      assert.equal(pad.head, 0);
+    });
+
+    it('getRevisionAuthor returns empty string for missing author', async function () {
+      await importEtherpad.setPadRaw(padId, JSON.stringify(makeOldExportNoAuthor()));
+      const pad = await padManager.getPad(padId);
+      assert.equal(await pad.getRevisionAuthor(0), '');
+    });
+  });
+
   describe('exportEtherpadAdditionalContent', function () {
     let hookBackup: Function;
 

